tests/src/Kernel/EmbargoAccessQueryTaggingAlterTest.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+namespace Drupal\Tests\embargo\Kernel;
+
+use Drupal\embargo\EmbargoInterface;
+use Drupal\Tests\islandora_test_support\Traits\DatabaseQueryTestTraits;
+
+/**
+ * Tests access queries are properly altered by embargo module.
+ *
+ * @group embargo
+ */
+class EmbargoAccessQueryTaggingAlterTest extends EmbargoKernelTestBase {
+  use DatabaseQueryTestTraits;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setUp(): void {
+    parent::setUp();
+
+    // Create two nodes one embargoed and one non-embargoed.
+    $embargoedNode = $this->createNode();
+    $this->createMedia($this->createFile(), $embargoedNode);
+    $this->embargo = $this->createEmbargo($embargoedNode);
+
+    $this->createNode();
+  }
+
+  /**
+   * Tests 'node_access' query alter, for nodes with embargo.
+   *
+   * Verifies that a user can view non-embargoed nodes only.
+   */
+  public function testEmbargoNodeQueryAlterAccess() {
+    $query = $this->generateNodeSelectAccessQuery($this->user);
+    $result = $query->execute()->fetchAll();
+    $this->assertCount(1, $result, 'User can only view non-embargoed node.');
+  }
+
+  /**
+   * Tests 'media_access' query alter, for embargoed node.
+   *
+   * Verifies that a user cannot view the media of an embargoed node.
+   */
+  public function testNodeEmbargoReferencedMediaAccessQueryAlterAccessDenied() {
+    $query = $this->generateMediaSelectAccessQuery($this->user);
+    $result = $query->execute()->fetchAll();
+    $this->assertCount(0, $result, 'Media of embargoed nodes cannot be viewed');
+  }
+
+  /**
+   * Tests 'file_access' query alter, for embargoed node.
+   *
+   * Verifies that a user cannot view the files of an embargoed node.
+   */
+  public function testNodeEmbargoReferencedFileAccessQueryAlterAccessDenied() {
+    $query = $this->generateFileSelectAccessQuery($this->user);
+    $result = $query->execute()->fetchAll();
+    $this->assertCount(1, $result, 'File of embargoed nodes cannot be viewed');
```

Comment:
Looking back at this, I'm not sure that this is correct. We only create the one embargoed file in the `::setUp()`, but we're checking that we can we can access a file, but there should only be the one that we do _not_ have access to?